### PR TITLE
Improve di container

### PR DESCRIPTION
**Pull request by yiiliveext**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #104

### CODE REVIEW

File: phpbench.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "bootstrap": "vendor/autoload.php",
-    "path": "tests/benchmark",
+    "path": "/tests/benchmark",
```
Comment (samdark) [inline phpbench.json]: That's not correct. Should be reverted.

File: src/Contracts/DeferredServiceProviderInterface.php [hunk 2]
```diff
@@ -16,4 +16,6 @@ interface DeferredServiceProviderInterface extends ServiceProviderInterface
      * @return bool whether service provider would register definition or not.
      */
     public function hasDefinitionFor(string $id): bool;
+
+    public function provides(): array;
```
Comment (samdark) [inline src/Contracts/DeferredServiceProviderInterface.php]:
```suggestion
    /**
      * @return string[] a list of IDs of services provided
      */
    public function provides(): array;
```

**Comment by samdark** [comment]
Thanks!

File: phpbench.json (hunk 1 above)
Comment (yiiliveext) [reply phpbench.json]: Pushed accidentally, otherwise it does not work on my PC.

Comment (samdark) [reply phpbench.json]: That is weird.

Comment (yiiliveext) [reply phpbench.json]: > That is weird.

No, just a phpbench bug, I will make PR for that.

Comment (yiiliveext) [reply phpbench.json]: Ups... The bug is already fixed, I've just run composer update :)

**Pull request merged**